fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingMerger.java

```diff
@@ -114,7 +114,9 @@ private CompletableFuture<Boolean> handleSuccess(final IndexDeferredMaintenanceC
         }
         // after a successful merge, reset the time quota and document count. They are unlikely to be applicable for the next merge.
         timeQuotaMillis = 0;
-        repartitionDocumentCount = 0;
+        if (repartitionDocumentCount > 0) {
+            repartitionDocumentCount = 0;
+        }
```

Comment:
* `shouldGiveRepartitionSecondChance` should have reset that. My typo. 
* Added an assertion to `TerribleRebalalce2ndChanceIndexMaintainer`. 
✅

Reply:
So, is there any point to the `if` block here?

Reply:
Yes. Unlike the regular indexer, that assumes correlation between "neighboring" records, this code assumes none for records that had been moved during repartitioning. Hence, if the number of moved records per single transaction was reduced in a previous repartitioning (because of failures), it will be reset back to default for the next one. 
@wshaib-apple is now working on code that will limit the total number of records to move in a single rebalance operation, his code will cause this function to return `AsyncUtil.READY_TRUE` (i.e. `hasMore`) when needed. So this reset is relevant for either a second chance or when only partial work was done.

Reply:
Note that if the repartitioning failed twice it will not be retried again. We should probably look into it in the future.